_posts/2019-11-20-bazel-1.2.0.md

```diff
@@ -0,0 +1,52 @@
+---
+layout: posts
+title: "Bazel 1.2"
+authors:
+  - philwo
+---
+
+We've just released [Bazel 1.2](https://github.com/bazelbuild/bazel/releases/tag/1.1.0)!
+
+Bazel 1.2 is intended to be backward-compatible with Bazel 1.0, however it contains a single incompatible change that was deemed a bug fix and thus eligible for inclusion.
+
+Please [report](https://github.com/bazelbuild/bazel/issues/new) any update problems.
+
+## Incompatible changes
+
+* Tree artifacts and regular artifact paths can no longer overlap. This could happen when a Starlark action creates a tree artifact and a file artifact underneath the tree. This triggered undefined behavior (e.g. what should a downstream action that depends on the tree artifact see?) and is now explicitly forbidden.
+
+## Android
+
+* `android_binary` and `android_library` can now depend on targets providing `CcInfos`. See [#9079](https://github.com/bazelbuild/bazel/issues/9079) for details.
+* Experimental support for d8 dexers (both standalone and incremental) is now available for use using `--nouse_workers_with_dexbuilder` `--define=android_incremental_dexing_tool=d8_dexbuilder`. See [#5150](https://github.com/bazelbuild/bazel/issues/5150) for details.
+* Added Desugar support for `FreezePeriod#<init>`.
+## Query
+
+* Fixed an aquery bug with handling malformed queries that crashes Bazel.
+
+## C++
+
+* The `cc_common.{compile,link}` APIs can now be used without passing the `--experimental_cc_skylark_api_enabled_packages` flag.
+* Incompatible flag [`--incompatible_validate_top_level_header_inclusions`](https://github.com/bazelbuild/bazel/issues/10047) has been added.
+* Fields on `CcLinkingOutputs` are now listed.
+
+## Platforms
+
+* `--target_platform_fallback` now also applies to exec / host configurations
+
+## Python
+
+* Added flag [`--incompatible_default_to_explicit_init_py`](https://github.com/bazelbuild/bazel/issues/10076) to switch the default value of [`legacy_create_init`](https://docs.bazel.build/versions/master/be/python.html#py_binary.legacy_create_init) to False. With this flag enabled, your `py_binary` and `py_test` targets will no longer behave as if empty `__init__.py` files were implicitly littered in your runfiles tree.
+
+## Documentation and UX
+
+* Bazel will now prevent idle sleep during test and build actions. Note that this does not affect screensavers and will not keep a laptop awake if the user forces sleep or closes the lid. This is purely to avoid idle sleeping when the user is not interacting with the device.
+* Added support for tracking suspensions (sleeps or SIGSTOP) on macOS.
+* BUILD/.bzl execution errors now cause execution to stop, even at top-level.
+* A list of log paths will now be provided in the build output.
+* Improved runfiles documentation.
```

Comment:
Add link to https://docs.bazel.build/versions/master/skylark/rules.html#runfiles

Reply:
Thank you!